vision/segmentation/plane_segmentation.go

```diff
@@ -223,7 +307,49 @@ func (pcps *pointCloudPlaneSegmentation) FindPlanes(ctx context.Context) ([]pc.P
 	var lastNonPlaneCloud pc.PointCloud
 	for {
 		lastNonPlaneCloud = nonPlaneCloud
-		smallerPlane, smallerNonPlaneCloud, err := SegmentPlane(ctx, nonPlaneCloud, pcps.nIterations, pcps.threshold)
+		smallerPlane, smallerNonPlaneCloud, err := SegmentPlane(ctx, nonPlaneCloud, pcps.nIterations, pcps.distanceThreshold)
+		if err != nil {
+			return nil, nil, err
+		}
+		planeCloud, err := smallerPlane.PointCloud()
+		if err != nil {
+			return nil, nil, err
+		}
+		if planeCloud.Size() <= pcps.minPoints {
+			// this cloud is not valid so revert to last
+			nonPlaneCloud = lastNonPlaneCloud
+			break
+		} else {
+			nonPlaneCloud = smallerNonPlaneCloud
+		}
+		planes = append(planes, smallerPlane)
+	}
+	return planes, nonPlaneCloud, nil
+}
+
+// FindGroundPlanes takes in a point cloud and outputs an array of the ground like planes and a point cloud of the leftover points.
+func (pcps *pointCloudPlaneSegmentation) FindGroundPlanes(ctx context.Context) ([]pc.Plane, pc.PointCloud, error) {
+	planes := make([]pc.Plane, 0)
+	var err error
+	plane, nonPlaneCloud, err := SegmentPlaneWRTGround(ctx, pcps.cloud, pcps.nIterations,
+		pcps.angleThreshold, pcps.distanceThreshold, pcps.normalVec)
+	if err != nil {
+		return nil, nil, err
+	}
+	planeCloud, err := plane.PointCloud()
+	if err != nil {
+		return nil, nil, err
+	}
+	if planeCloud.Size() <= pcps.minPoints {
+		return planes, pcps.cloud, nil
```

Comment:
you can return nil plane if there is not a ground plane in the image. It will be up to the function getting the result to check and figure out what to do if there are no ground planes.